File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] visited = new boolean[n];
        
        for(int i=0; i<n; i++) {
        	for(int j=0; j<n; j++) {
        		if(computers[i][j] == 1 && visited[j] == false) {
        			count(n, computers, visited, i);
        			answer++;
        		}
        	}
        }
        
        return answer;
    }
    
    public static void count(int n, int[][] computers, boolean[] visited, int i) {
		Queue<Integer> q= new LinkedList<>();
		q.offer(i);
		
		while(!q.isEmpty()) {
			int index = q.poll();
			for(int j=0; j<n; j++) {
				if(computers[index][j] == 1 && visited[j] == false) {
					visited[j] = true;
					q.offer(j);
				}
			}
		}
	}
    
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

answer = 0

def solution(n, computers):
    visited = [False] * n

    for i in range(0, n):
        if not visited[i]:
            BFS(n, computers, i, visited)

    return answer

def BFS(n, computers, index, visited):
    global answer
    queue = deque()
    queue.append(computers[index])
    visited[index] = True

    answer += 1

    while queue:
        temp = queue.popleft()
        for j in range(0, n):
            if j != index and temp[j] == 1 and visited[j] == False:
                visited[j] = True
                queue.append(computers[j])